webapi/core.py

```diff
@@ -1,20 +1,95 @@
 import logging
 
+from deluge import component
 from deluge.plugins.pluginbase import CorePluginBase
+from deluge.ui.web.json_api import JSON
+from twisted.web import http, server
+import deluge.configmanager
+from deluge.core.rpcserver import export
 
 
 LOGGER = logging.getLogger(__name__)
 
+DEFAULT_PREFS = {
+    "enable_cors"         : False,
+    "allowed_origin"      : []
+}
 
 class Core(CorePluginBase):
 
     def enable(self):
         """"""
         LOGGER.info('Enabling WebAPI plugin CORE ...')
+        self.JSON_instance = component.get('JSON')
+        self.patched = False
+        self.config = deluge.configmanager.ConfigManager("webapi.conf", DEFAULT_PREFS)
+        if (self.config["enable_cors"] == True):
```

Comment:
`if self.config['enable_cors']:` would do. Here and elsewhere no need in explicit checks.